src/Microsoft.VisualBasic/tests/StringsTests.cs

```diff
@@ -165,21 +165,141 @@ public void Filter_MultipleElements(string[] source, string match, string[] incl
         [Fact]
         public void Filter_Objects_WhenObjectCannotBeConvertedToString_ThrowsArgumentOutOfRangeException()
         {
-            var source = new object[] { typeof(object) };
-            var match = "a";
+            object[] source = new object[] { typeof(object) };
+            string match = "a";
 
             AssertExtensions.Throws<ArgumentException>("Source", null, () => Strings.Filter(source, match));
         }
 
         [Theory]
         [InlineData(new object[] { 42 }, "42", new string[] { "42" }, new string[] { })]
         [InlineData(new object[] { true }, "True", new string[] { "True" }, new string[] { })]
-        public void Objects(object[] source, string match, string[] includeExpected, string[] excludeExpected)
+        public void Filter_Objects(object[] source, string match, string[] includeExpected, string[] excludeExpected)
         {
             Assert.Equal(includeExpected, Strings.Filter(source, match, Include: true));
             Assert.Equal(excludeExpected, Strings.Filter(source, match, Include: false));
         }
 
+        [Theory]
+        [MemberData(nameof(InStr_TestData_NullsAndEmpties))]
+        [MemberData(nameof(InStr_FromBegin_TestData))]
+        public void InStr_FromBegin(string string1, string string2, int expected)
+        {
+            Assert.Equal(expected, Strings.InStr(string1, string2));
+            Assert.Equal(expected, Strings.InStr(1, string1, string2));
+        }
+
+        [Theory]
+        [MemberData(nameof(InStr_TestData_NullsAndEmpties))]
+        [MemberData(nameof(InStr_FromWithin_TestData))]
+        public void InStr_FromWithin(string string1, string string2, int expected)
+        {
+            int startPos = 2;
+            Assert.Equal(expected, Strings.InStr(startPos, string1, string2));
+        }
+
+        [Theory]
+        [InlineData("A", "a", 0)]
+        [InlineData("Aa", "a", 2)]
+        public void InStr_BinaryCompare(string string1, string string2, int expected)
+        {
+            Assert.Equal(expected, Strings.InStr(string1, string2, CompareMethod.Binary));
+            Assert.Equal(expected, Strings.InStr(1, string1, string2, CompareMethod.Binary));
+        }
+
+        [Theory]
+        [InlineData("A", "a", 1)]
+        [InlineData("Aa", "a", 1)]
+        public void InStr_TextCompare(string string1, string string2, int expected)
+        {
+            Assert.Equal(expected, Strings.InStr(string1, string2, CompareMethod.Text));
+            Assert.Equal(expected, Strings.InStr(1, string1, string2, CompareMethod.Text));
+        }
+
+        [Theory]
+        [InlineData(2)]
+        [InlineData(3)]
+        public void InStr_WhenStartGreatherThanLength_ReturnsZero(int start)
+        {
+            string string1 = "a";
+            string string2 = "a";
+            int expected = 0;
```

Comment:
no need to define these variable, you can directly provide them in function call